api/envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.api_key_auth.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/sensitive.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.api_key_auth.v3";
+option java_outer_classname = "ApiKeyAuthProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/api_key_auth/v3;api_key_authv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: APIKey Auth]
+// APIKey Auth :ref:`configuration overview <config_http_filters_api_key_auth>`.
+// [#extension: envoy.filters.http.api_key_auth]
+
+// API Key HTTP authentication.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//    authentication_header: "X-API-KEY"
+//    keys:
+//      inline_string: |-
+//        clientID1:apiKey1
+//        clientID2:apiKey2
+//
+message APIKeyAuth {
+  // keys used to authenticate the client.
+  // It should be a map of clientID to apiKey. 
+  // The clientID serves solely for identification purposes and isn't used for authentication.
+  config.core.v3.DataSource keys = 1 [(udpa.annotations.sensitive) = true];
+
+  // The header name to fetch the key.
+  // If multiple values are present in the given header, the filter rejects the request.
+  // Only one of authentication_header, authentication_query, or authentication_cookie should be set.
+  string authentication_header = 2;
+  // The query parameter name to fetch the key.
+  // Only one of authentication_header, authentication_query, or authentication_cookie should be set.
+  string authentication_query = 3;
+  // The cookie name to fetch the key.
+  // Only one of authentication_header, authentication_query, or authentication_cookie should be set.
+  string authentication_cookie = 4;
+}
```

Comment:
By the way, I think may be we can use the `APIKeyAuth` as our per route config directly. 

First, it would simplify the implementation because the filter config and route level config could shared completely same implementation.
Second, it make it is possible to use different key positions for different host/route.

Reply:
Good point, will update like that along with the implementation

Reply:
Done in https://github.com/envoyproxy/envoy/pull/36709/commits/1f5cfadf9e3e7d369bae76b94181a02709b69fe7